### DISCUSSION TITLE
[Chapter 01] MSA의 Transaction

### DISCUSSION THREAD

**Discussion by ahngilwoong** [cowaine/spring-boot-for-msa#4] [Chapter 01. Microservices Architecture]
1. 책에 나온 MSA 환경에서의 트랜잭션 처리 방법에 대해 자세히 의견을 나눠보고 싶습니다.

 - 책에서는 MSA 환경은 서버들이 각각 독립되어있는 환경이라 트랜잭션 처리가 굉장히 어렵다 라고 나와있습니다.
 - 그에 대한 해결책으로 커넥션 풀을 이용할 수 있고 시간?을 맞춰야 한다고 하는데 이게 정확히 무슨 뜻인지?
 - 트랜잭션 처리에 대해서 REST API로는 신뢰성이 없다고 하는데 Kafka 같은 메시징 큐를 이용하여 처리하는 것도 하나의 방법이 될지?
 - 현재 MSA환경에서 트랜잭션을 처리하는 표준? 같은 방법이 무엇인지?
 - 주문? 등의 예시를 가정해서 MSA 환경에서 트랜잭션 하는 Flow를 한번 다같이 정리해보면 좋을 것 같습니다.

**Comment by ahngilwoong** [comment]
위 질문들은 진도가 나아가며 궁금증이 해결된 부분들이 다수입니다.

**Reply by corock** [reply]
이번 스터디를 통해 성장하신 건가요?

**Reply by ahngilwoong** [reply]
맞습니다 ㅎㅎ
